common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -297,17 +301,21 @@ def transcript(self, request, dispatch):
                 # if no translation is required
                 response = self.get_static_transcript(request, transcripts)
                 if response.status_code == 404 and feature_enabled:
-                    # Try to get transcript from edx-val as a last resort.
-                    transcript = get_video_transcript_content(self.edx_video_id, self.transcript_language)
-                    if transcript:
-                        transcript_conversion_props = dict(transcript, output_format=Transcript.SJSON)
-                        transcript = convert_video_transcript(**transcript_conversion_props)
-                        response = Response(
-                            transcript['content'],
-                            headerlist=[('Content-Language', self.transcript_language)],
-                            charset='utf8',
-                        )
-                        response.content_type = Transcript.mime_types[Transcript.SJSON]
+                    content, filename, mimetype = get_transcript_from_val(
+                        self.edx_video_id,
```

Comment:
we should clean `edx_video_id` here too.

Reply:
We are not going to merge this into the master, it'll be good to minimize the changes to transcript util interface, and move this work into child PR

Reply:
> we should clean edx_video_id here too.

`get_transcript_from_val` calls `get_video_transcript_content` and cleaning is happening there.